fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/impl/tool/attribute/ToolManagerImpl.java

```diff
@@ -253,8 +258,9 @@ default ActionResult isEffectiveOn(Tag<Item> tag, BlockState state, ItemStack st
 		 * @param user  the user involved in breaking the block, null if not applicable.
 		 * @return the result of mining speed.
 		 */
+		@NotNull
 		default TypedActionResult<Float> getMiningSpeedMultiplier(Tag<Item> tag, BlockState state, ItemStack stack, @Nullable LivingEntity user) {
-			return null;
+			return TypedActionResult.pass(1.0F);
```

Comment:
Was there a reason this was null?

Reply:
There is actually an Objects#requireNonNull check originally anyways, this was just a wrong default value.